src/vmshepherd/errors.py

```diff
@@ -0,0 +1,58 @@
+class VmShepherdError(Exception):
+    message = None
+    details = None
+
+    def __init__(self, message, details=None):
+        super().__init__(self.message)
+        self.message = message
+        self.details = details if details else ''
+        assert self.message is not None
+
+    def __str__(self):
+        if self.details:
+            return 'VmShepherdError: %s, details=%s' % (self.message, self.details)
```

Comment:
using `%s` is so Python2 way of doing things, `VmShepherd` runs on Python 3.6. One of it's cool features is [fstrings](https://docs.python.org/3/whatsnew/3.6.html#whatsnew36-pep498). How about using them? :)